scripts/pre-commit.git-lint.sh

```diff
@@ -14,8 +14,13 @@
 # limitations under the License.
 
 # First part return the files being commited, excluding deleted files.
-git diff-index -z --cached HEAD --name-only --diff-filter=ACMRTUXB |
-xargs --null --no-run-if-empty git lint;
+DIFF_FILES=$(git diff-index -z --cached HEAD --name-only --diff-filter=ACMRTUXB)
```

Comment:
Bash does not store new lines nor `\0`, so the above won't work. See http://stackoverflow.com/questions/6570531/assign-string-containing-null-character-0-to-a-variable-in-bash#answer-22985397

Apparently the only bullet proof solution is to save the result to a file, but that has the complications of deleting the temporary file even in case of errors.

Another alternative is to run the command twice. The first time to check whether the output is empty. And the second piped to xargs.